include/rdycore.h.in

```diff
@@ -26,10 +26,23 @@ PETSC_EXTERN PetscErrorCode RDyOnFinalize(void (*)(void));
 PETSC_EXTERN PetscErrorCode RDyFinalize(void);
 PETSC_EXTERN PetscBool RDyInitialized(void);
 
-// RDycore setup/run/breakdown
+// RDycore setup/breakdown
 PETSC_EXTERN PetscErrorCode RDyCreate(MPI_Comm, const char*, RDy*);
 PETSC_EXTERN PetscErrorCode RDySetup(RDy);
-PETSC_EXTERN PetscErrorCode RDyRun(RDy);
 PETSC_EXTERN PetscErrorCode RDyDestroy(RDy*);
 
+// time integration
+PETSC_EXTERN PetscErrorCode RDyAdvance(RDy);
+
+// Accessing data
+PETSC_EXTERN PetscBool      RDyFinished(RDy);
+PETSC_EXTERN PetscErrorCode RDyGetTime(RDy, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetTimeStep(RDy, PetscReal*);
+PETSC_EXTERN PetscErrorCode RDyGetStep(RDy, PetscInt*);
+PETSC_EXTERN PetscErrorCode RDyGetCouplingInterval(RDy, PetscReal*);
```

Comment:
How about also adding an `RDySetCouplingInterval()`? This will allow E3SM to override the coupling interval specified in the `.yaml` file.